### ISSUE TITLE
[Feature]: Support cloudflare AI Gateway - universal link

### ISSUE THREAD

**Issue by shuther** [BerriAI/litellm#1158]
### The Feature

CF has a free offer for this gateway but there are multiple url for each endpoint and one universal entry point [CF link](https://developers.cloudflare.com/ai-gateway). The idea is to connect to this universal entry point, and not to specific ones.

### Motivation, pitch

this gateway should be seen as an observer third party.

**Comment by ghost** [comment]
oh - what does 'free offer' mean? @shuther

**Comment by shuther** [comment]
[some limits](https://developers.cloudflare.com/workers-ai/platform/limits/) but while [future pricing](https://blog.cloudflare.com/workers-ai) has been shared, it is not yet available; we don't pay for the usage as it is not considered production ready. So good for testing.

**Comment by Manouchehri** [comment]
That's the pricing for Workers AI, not AI Gateway. AI Gateway will _likely_ be free (or close to it) forever.

**Comment by shuther** [comment]
yes, but connecting to the API Gateway is interesting as an observer on the edge (but their approach in term of entry point is not practical).
And connecting to the cloudflare workers AI is interesting in term of offer.
Therefore, connecting from litellm to Cloudflare API Gateway seems a compelling offer; or do you have a concern with it?

**Comment by Manouchehri** [comment]
Oh I love AI Gateway; I think I’m the first LiteLLM user who started to use AI Gateway with Azure OpenAI on LiteLLM. It’s super nice for troubleshooting.

What would LiteLLM gain by using the universal endpoint though? LiteLLM already has fallbacks.

**Comment by shuther** [comment]
> Oh I love AI Gateway; I think I’m the first LiteLLM user who started to use AI Gateway with Azure OpenAI on LiteLLM. It’s super nice for troubleshooting.
> 
> 
> 
> What would LiteLLM gain by using the universal endpoint though? LiteLLM already has fallbacks. 

So you are using litellm->ai gateway->azure?
I would prefer to have the flexibility to change the model from the application (mix between open ai and mistral) so I would need to leverage the universal endpoint to manage the connection (fall back is more cloudflare vs open router)
